src/metabase/query_processor/util/nest_query.clj

```diff
@@ -19,6 +19,24 @@
    [metabase.util :as u]
    [metabase.util.log :as log]))
 
+(defn- all-fields-for-table [table-id]
+  (->> (lib.metadata/fields (qp.store/metadata-provider) table-id)
+       (remove #(#{:sensitive :retired} (:visibility-type %)))
+       (map (fn [field]
+              [:field (u/the-id field) nil]))))
+
+(defn- add-all-fields
```

Comment:
This might be a good place to explain why this function is needed, basically your PR's description.